azurerm/internal/services/apimanagement/api_management_identity_provider_aadb2c_resource.go

```diff
@@ -0,0 +1,228 @@
+
+			"client_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsUUID,
+			},
+
+			"client_secret": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			// For AADB2C identity providers, `allowed_tenants` must specify exactly one tenant
+			"allowed_tenant": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"signin_tenant": {
+				Type:     schema.TypeString,
+				Required: true,
+				// B2C tenant domains can be customized, and GUIDs might work here too
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"authority": {
+				Type:     schema.TypeString,
+				Required: true,
+				// B2C login domains can be customized and don't necessarily end in b2clogin.com
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"signup_policy": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"signin_policy": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"profile_editing_policy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"password_reset_policy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func resourceArmApiManagementIdentityProviderAADB2CCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.IdentityProviderClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+
+	clientID := d.Get("client_id").(string)
+	clientSecret := d.Get("client_secret").(string)
+
+	allowedTenant := d.Get("allowed_tenant").(string)
+	signinTenant := d.Get("signin_tenant").(string)
+	authority := d.Get("authority").(string)
+	signupPolicy := d.Get("signup_policy").(string)
+
+	signinPolicy := d.Get("signin_policy").(string)
+	profileEditingPolicy := d.Get("profile_editing_policy").(string)
+	passwordResetPolicy := d.Get("password_reset_policy").(string)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, serviceName, apimanagement.AadB2C)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of existing Identity Provider %q (API Management Service %q / Resource Group %q): %s", apimanagement.AadB2C, serviceName, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_api_management_identity_provider_aadb2c", *existing.ID)
```

Comment:
rather than using the API ID here, can we use the ID from the Parser/Formatter here?